File: constants/apiEndpoints.ts
/**
 * ============================================================================
 * API ENDPOINTS
 * ============================================================================
 *
 * Enum containing all API endpoint paths.
 * Use these instead of hardcoding paths in controllers.
 *
 * Usage:
 *   import { EndpointsEnum } from "@constants/apiEndpoints";
 *
 *   const response = await this.send("get", EndpointsEnum.Users, headers);
 *
 * To add a new endpoint:
 *   1. Add the path here with a descriptive name
 *   2. Use it in your controller
 * ============================================================================
 */

export enum EndpointsEnum {
  // ===========================================================================
  // AUTHENTICATION
  // ===========================================================================

  /** User login */
  Login = "/api/gateway/auth/login",

  /** User logout */
  Logout = "/api/gateway/auth/logout",

  /** Get current user info */
  AuthInfo = "/api/gateway/auth/info",

  /** User signup */
  SignUp = "/api/gateway/auth/sign-up",

  /** Password reset request */
  PasswordReset = "/api/gateway/auth/password/reset",

  /** Change password */
  AuthPassword = "/api/gateway/auth/password",

  /** Invite user */
  AuthInvite = "/api/gateway/auth/invite",

  /** Delete user account */
  AuthDelete = "/api/gateway/auth/delete",

  // ===========================================================================
  // USER MANAGEMENT
  // ===========================================================================

  /** Get user details */
  UserInfo = "/api/gateway/user/info",

  /** User operations */
  User = "/api/gateway/user/",

  /** Get all users */
  Users = "/api/gateway/statistics/tenant/users",

  /** User details */
  UsersDetails = "/api/gateway/user/details",

  /** Remove user */
  UserRemove = "/api/gateway/auth/delete?userId=",

  /** Restore user */
  UserRestore = "/api/gateway/user/restore",

  // ===========================================================================
  // USER GROUPS
  // ===========================================================================

  /** User groups management */
  UserGroupsManage = "/api/gateway/user/groups",

  /** Single user group */
  UserGroup = "/api/gateway/user/group",

  // ===========================================================================
  // PRESENTATIONS/CONTENT
  // ===========================================================================

  /** Presentation API base */
  PresentationApi = "/api/gateway/presentations/",

  /** Presentations page */
  Presentations = "/presentations/",

  /** Presentation state */
  PresentationsState = "/api/gateway/presentations/state",

  /** Save presentation */
  PresentationsSave = "/api/gateway/presentations/save",

  /** Confirm presentation */
  PresentationsConfirm = "/api/gateway/presentations/confirm",

  /** Delete presentation(s) */
  PresentationsDelete = "/api/gateway/presentations/?presentationIds=",

  /** Get presentation by ID */
  PresentationsId = "/api/gateway/presentations/v1/presentation",

  /** Copy presentation */
  PresentationCopy = "/api/gateway/presentations/copy",

  /** Comments on presentation */
  Comments = "/api/gateway/comment",

  // ===========================================================================
  // FILES & MEDIA
  // ===========================================================================

  /** Files API */
  Files = "/api/gateway/files/",

  /** File info */
  FilesInfo = "/api/gateway/files/v1",

  /** File ID for presentation */
  FilesId = "/api/gateway/presentations/files/",

  /** Delete file */
  FilesDelete = "/api/gateway/files/delete/",

  /** Single file info */
  FileInfo = "/api/gateway/files/file",

  /** Upload file stream */
  UploadFileStream = "/api/gateway/files/upload/stream/",

  /** Media metadata */
  MediaMetaData = "/api/gateway/files/media-metadata",

  /** Media library page */
  MediaLibrary = "/manage/assets",

  // ===========================================================================
  // ACCESS CONTROL
  // ===========================================================================

  /** Single group */
  Group = "/api/gateway/access-control/group",

  /** All groups */
  Groups = "/api/gateway/access-control/groups",

  /** Group by ID */
  GroupId = "/api/gateway/access-control/group?groupId=",

  /** Presentation access control */
  PresentationsApi = "/api/gateway/access-control/presentation",

  // ===========================================================================
  // EXPLORE/PORTAL
  // ===========================================================================

  /** Search presentations in explore */
  ExplorePresentations = "/api/gateway/search/presentations/explore",

  /** Share presentation */
  PresentationShare = "/api/gateway/portal/presentation",

  /** Recommendations */
  Recommendations = "/api/gateway/portal/recommendations",

  // ===========================================================================
  // STATISTICS
  // ===========================================================================

  /** Presentation views */
  PresentationViewed = "/api/gateway/presentations/viewed/",

  /** User views statistics */
  StatisticsUserViews = "/api/gateway/statistics/user/views",

  /** Views by host */
  StatisticsUserViewsByHost = "/api/gateway/statistics/user/viewsByHost",

  /** Views by presentation */
  StatisticsUserViewsByPresentation = "/api/gateway/statistics/user/viewsByPresentation",

  /** Duration statistics */
  StatisticsUserDuration = "/api/gateway/statistics/user/duration",

  // ===========================================================================
  // TENANT
  // ===========================================================================

  /** Tenant details */
  TenantDetails = "/api/gateway/tenant/details",

  // ===========================================================================
  // FORMS & CERTIFICATES
  // ===========================================================================

  /** Forms page */
  Forms = "/manage/forms",

  /** Create form */
  FormCreate = "/api/gateway/form/form",

  /** Delete form */
  FormDelete = "/api/gateway/form/delete?formId=",

  /** Form info */
  FormInfo = "/api/gateway/form/info?formId=",

  /** All forms */
  FormAll = "/api/gateway/form/all",

  /** Attach form to presentation */
  FormAttachToPresentation = "/api/gateway/presentations/feedback-form",

  /** Submit form */
  FormSubmit = "/api/gateway/form/submit",

  /** Certificate template */
  CertificateOfForm = "/api/gateway/form/template/",

  /** Download certificate */
  CertificateDownload = "/api/gateway/form/certificate",

  /** All certificates */
  CertificateAll = "/api/gateway/form/certificate/all",

  // ===========================================================================
  // EXTERNAL SERVICES
  // ===========================================================================

  /** Get public IP */
  GetMyIp = "https://api.ipify.org?format=json",

  // ===========================================================================
  // ADD MORE ENDPOINTS HERE
  // ===========================================================================
}

File: python_playwright/constants/api_endpoints.py
from enum import StrEnum


class Endpoints(StrEnum):
    # AUTHENTICATION
    LOGIN = "/api/gateway/auth/login"
    LOGOUT = "/api/gateway/auth/logout"
    AUTH_INFO = "/api/gateway/auth/info"
    SIGN_UP = "/api/gateway/auth/sign-up"
    PASSWORD_RESET = "/api/gateway/auth/password/reset"
    AUTH_PASSWORD = "/api/gateway/auth/password"
    AUTH_INVITE = "/api/gateway/auth/invite"
    AUTH_DELETE = "/api/gateway/auth/delete"

    # USER MANAGEMENT
    USER_INFO = "/api/gateway/user/info"
    USER = "/api/gateway/user/"
    USERS = "/api/gateway/statistics/tenant/users"
    USERS_DETAILS = "/api/gateway/user/details"
    USER_REMOVE = "/api/gateway/auth/delete?userId="
    USER_RESTORE = "/api/gateway/user/restore"

    # USER GROUPS
    USER_GROUPS_MANAGE = "/api/gateway/user/groups"
    USER_GROUP = "/api/gateway/user/group"

    # PRESENTATIONS/CONTENT
    PRESENTATION_API = "/api/gateway/presentations/"
    PRESENTATIONS = "/presentations/"
    PRESENTATIONS_STATE = "/api/gateway/presentations/state"
    PRESENTATIONS_SAVE = "/api/gateway/presentations/save"
    PRESENTATIONS_CONFIRM = "/api/gateway/presentations/confirm"
    PRESENTATIONS_DELETE = "/api/gateway/presentations/?presentationIds="
    PRESENTATIONS_ID = "/api/gateway/presentations/v1/presentation"
    PRESENTATION_COPY = "/api/gateway/presentations/copy"
    COMMENTS = "/api/gateway/comment"

    # FILES & MEDIA
    FILES = "/api/gateway/files/"
    FILES_INFO = "/api/gateway/files/v1"
    FILES_ID = "/api/gateway/presentations/files/"
    FILES_DELETE = "/api/gateway/files/delete/"
    FILE_INFO = "/api/gateway/files/file"
    UPLOAD_FILE_STREAM = "/api/gateway/files/upload/stream/"
    MEDIA_METADATA = "/api/gateway/files/media-metadata"
    MEDIA_LIBRARY = "/manage/assets"

    # ACCESS CONTROL
    GROUP = "/api/gateway/access-control/group"
    GROUPS = "/api/gateway/access-control/groups"
    GROUP_ID = "/api/gateway/access-control/group?groupId="
    PRESENTATIONS_API_ACCESS = "/api/gateway/access-control/presentation"

    # EXPLORE/PORTAL
    EXPLORE_PRESENTATIONS = "/api/gateway/search/presentations/explore"
    PRESENTATION_SHARE = "/api/gateway/portal/presentation"
    RECOMMENDATIONS = "/api/gateway/portal/recommendations"

    # STATISTICS
    PRESENTATION_VIEWED = "/api/gateway/presentations/viewed/"
    STATISTICS_USER_VIEWS = "/api/gateway/statistics/user/views"
    STATISTICS_USER_VIEWS_BY_HOST = "/api/gateway/statistics/user/viewsByHost"
    STATISTICS_USER_VIEWS_BY_PRESENTATION = "/api/gateway/statistics/user/viewsByPresentation"
    STATISTICS_USER_DURATION = "/api/gateway/statistics/user/duration"

    # TENANT
    TENANT_DETAILS = "/api/gateway/tenant/details"

    # FORMS & CERTIFICATES
    FORMS = "/manage/forms"
    FORM_CREATE = "/api/gateway/form/form"
    FORM_DELETE = "/api/gateway/form/delete?formId="
    FORM_INFO = "/api/gateway/form/info?formId="
    FORM_ALL = "/api/gateway/form/all"
    FORM_ATTACH_TO_PRESENTATION = "/api/gateway/presentations/feedback-form"
    FORM_SUBMIT = "/api/gateway/form/submit"
    CERTIFICATE_OF_FORM = "/api/gateway/form/template/"
    CERTIFICATE_DOWNLOAD = "/api/gateway/form/certificate"
    CERTIFICATE_ALL = "/api/gateway/form/certificate/all"

    # EXTERNAL SERVICES
    GET_MY_IP = "https://api.ipify.org?format=json"
